### PR TITLE
New catalogue data

### PR DESCRIPTION
**Pull request by gestchild**
References #4661

Displays the new data that's been added to the catalogue API on the works page (behind a toggle)

### CODE REVIEW

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 1]
```diff
@@ -196,10 +221,27 @@ const WorkDetails = ({
     work.physicalDescription ||
     work.lettering ||
     work.genres.length > 0 ||
-    work.language
+    work.language ||
+    (showAdditionalCatalogueData &&
+      (work.alternativeTitle ||
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: `alternativeTitles`

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 2]
```diff
@@ -29,6 +29,27 @@ import { clientSideSearchParams } from '@weco/common/services/catalogue/search-p
...
 import PaletteSimilarityBox from '../PaletteSimilarityBox/PaletteSimilarityBox';
 
+type TimeProps = {|
+  hours: string,
+  minutes: string,
+  seconds: string,
+|};
+
+function doubleDigits(n) {
+  return n > 9 ? `${n}` : `0${n}`;
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: Could use `padStart` and add it to polyfil for ie?

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 3]
```diff
@@ -29,6 +29,27 @@ import { clientSideSearchParams } from '@weco/common/services/catalogue/search-p
...
+  const seconds = milliseconds / 1000;
+  const closestSeconds = Math.round(seconds);
+  const minutes = Math.floor(closestSeconds / 60);
+  const hours = Math.floor(minutes / 60);
+  return {
+    hours: doubleDigits(hours),
+    minutes: doubleDigits(minutes % 60),
+    seconds: doubleDigits(closestSeconds % 60),
+  };
+}
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: Could we not use `moment.duration` for readability?

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 4]
```diff
@@ -196,10 +221,27 @@ const WorkDetails = ({
     work.physicalDescription ||
     work.lettering ||
     work.genres.length > 0 ||
-    work.language
+    work.language ||
+    (showAdditionalCatalogueData &&
+      (work.alternativeTitle ||
+        work.contributors ||
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: Contributors is already added?

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 5]
```diff
@@ -196,10 +221,27 @@ const WorkDetails = ({
...
+        work.dissertation ||
+        work.edition ||
+        work.duration ||
+        work.notes ||
+        work.contents ||
+        work.credits))
   ) {
     WorkDetailsSections.push(
       <WorkDetailsSection headingText="About this work">
+        {showAdditionalCatalogueData && work.description && (
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: Why `work.description`?

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 6]
```diff
@@ -239,11 +305,36 @@ const WorkDetails = ({
           />
         )}
 
-        {work.lettering && (
+        <MetaUnit
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: ?

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 7]
```diff
@@ -29,6 +29,27 @@ import { clientSideSearchParams } from '@weco/common/services/catalogue/search-p
...
+  hours: string,
+  minutes: string,
+  seconds: string,
+|};
+
+function doubleDigits(n) {
+  return n > 9 ? `${n}` : `0${n}`;
+}
+
+function convertMS(milliseconds: number): TimeProps {
```
Comment (davidpmccormick) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: Nitpick – milliseconds is one word, so don't feel like the 's' deserves capping up. I'd also consider spelling it out (and possibly adding what they're being converted to). Triple nit-pick, if you like!

File: catalogue/webapp/components/WorkDetails/WorkDetails.js (hunk 2 above)
Comment (davidpmccormick) [reply catalogue/webapp/components/WorkDetails/WorkDetails.js]: I wasn't even aware `padStart` was a thing. Nice.

File: catalogue/webapp/components/WorkDetails/WorkDetails.js (hunk 3 above)
Comment (gestchild) [reply catalogue/webapp/components/WorkDetails/WorkDetails.js]: moment.duration doesn't provide the formatting we need, but we can just use moment.utc and format that, which will work for anything up to 24hrs in duration

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 8]
```diff
@@ -192,14 +198,31 @@ const WorkDetails = ({
...
-    work.language
+    work.language ||
+    (showAdditionalCatalogueData &&
+      (work.alternativeTitles ||
+        work.dissertation ||
+        work.edition ||
+        work.duration ||
+        work.notes ||
+        work.contents ||
+        work.credits))
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js]: Bikeshedding - but what exactly are we trying to say with this logic again?

**Pull request merged**